dragonfly_schema/window_parameter.py

```diff
@@ -90,3 +91,98 @@ class RepeatingWindowRatio(BaseModel):
         description='An optional number to create a single vertical separation '
             'between top and bottom windows.'
     )
+
+
+class DetailedRectangularWindows(BaseModel):
+    """Several rectangular windows, defined by origin, width and height."""
+
+    type: constr(regex='^DetailedRectangularWindows$') = 'DetailedRectangularWindows'
+
+    origins: List[List[float]] = Field(
+        ...,
+        description='An array of 2D points within the plane of the wall for the origin '
+            'of each window. Each point should be a list of 2 (x, y) values. The '
+            'wall plane is assumed to have an origin at the first point of the wall '
+            'segment and an X-axis extending along the length of the segment. The '
+            'wall plane Y-axis always points upwards. Therefore, both X and Y '
+            'values of each origin point should be positive.'
+    )
+
+    widths: List[float] = Field(
+        ...,
+        description='An array of postive numbers for the window widths. '
+            'The length of this list must match the length of the origins.'
+    )
+
+    heights: List[float] = Field(
+        ...,
+        description='An array of postive numbers for the window heights. '
+            'The length of this list must match the length of the origins.'
+    )
+
+    @validator('origins')
+    def check_num_items(cls, v):
+        for i in v:
+            assert len(i) == 2, 'Number of floats must be 2 for (x, y).'
+            assert i[0] > 0, 'X value for WindowParameter origin must be positive.'
+            assert i[1] > 0, 'Y value for WindowParameter origin must be positive.'
+        return v
+
+    @validator('widths')
+    def check_positive_widths(cls, v):
+        for i in v:
+            assert i > 0, 'Window width must be positive.'
+        return v
+
+    @validator('heights')
+    def check_positive_heights(cls, v):
+        for i in v:
+            assert i > 0, 'Window height must be positive.'
+        return v
+
+    @root_validator
+    def check_aligned_lists(cls, values):
+        "Ensure length of origins, widths and heights match."
+        origins = values.get('origins')
+        widths = values.get('widths')
+        heights = values.get('heights')
+
+        assert len(origins) == len(widths) == len(heights), 'Length of ' \
+            'DetailedRectangularWindows origins, widths, and heights ' \
+            'must match. origins: {}, widths: {}, heights: {}'.format(
+                len(origins), len(widths), len(heights))
+        
+        return values
+
+
+class DetailedWindows(BaseModel):
```

Comment:
This can also be just `Windows`.

Reply:
That might be a bit dangerous.  It's not only a very vague class name but I feel intuitively that having a class share the same name as an operating system could result in some weird situations on some machines.  If you don't mind, I'd rather leave this as `DetailedWindows`.